app/models/manifest.rb

```diff
@@ -5,6 +5,12 @@
 class Manifest < ActiveRecord::Base
   validates :tracking_number, presence: true
 
+  before_validation :set_tracking_number, on: :create
+
+  def set_tracking_number
```

Comment:
where should the column number be set then? It needs to happen before the INSERT.